desc/compute/utils.py

```diff
@@ -691,15 +690,23 @@ def surface_integrals(grid, q=jnp.array([1]), surface_label="rho", max_surface=F
     # 'nodes', so that the duplicate surface has nodes with the same
     # surface label and is treated as one, like in the previous paragraph.
 
-    integrals = jnp.histogram(nodes_modulo, bins=bins, weights=ds * q)[0]
-    if has_endpoint_dupe:
+    def integrate_component(integrands):
+        integrals = jnp.histogram(nodes_modulo, bins=bins, weights=integrands)[0]
         # By modulating nodes, we 'moved' all the area from
         # surface_label=max_surface_val to the surface_label=0 surface.
         # With the correct value of the integral on this surface stored
         # in integrals[0], we copy it to the duplicate surface at integrals[-1].
-        assert integrals[-1] == 0
-        integrals = put(integrals, -1, integrals[0])
-    return expand(grid, integrals, surface_label)
+        if has_endpoint_dupe:
+            # could assert integrals[-1] == 0 but it offends jax
```

Comment:
Prior to coding with desc, I would use assertion statements for checks which I didn't want running in production code. For example, on a function designed to be logarithmic time worst case, an assert statement could call a function which checks program correctness in linear time. On checks I did want running in production code, I would throw errors (`raise AssertionError`). An example of this would be guards in functions to prevent bad user input. Then running python with the `python -O` (means optimized mode) would disable assertions but keep the errors.

I think the default for desc module is without the `-O`. If we want to uncomment this, I'll need to do that `jax.cond` thing or some trick with jnp.where

Reply:
there is a jax-ish way to do that but still experimental and probably not ideal:
https://jax.readthedocs.io/en/latest/_autosummary/jax.experimental.checkify.checkify.html